packages/grafana-data/src/types/transformations.ts

```diff
@@ -1,17 +1,17 @@
+import { Observable } from 'rxjs';
+
 import { DataFrame, Field } from './dataFrame';
 import { RegistryItemWithOptions } from '../utils/Registry';
 
 /**
  * Function that transform data frames (AKA transformer)
  */
-export type DataTransformer = (data: DataFrame[]) => DataFrame[];
-
 export interface DataTransformerInfo<TOptions = any> extends RegistryItemWithOptions {
   /**
    * Function that configures transformation and returns a transformer
    * @param options
    */
-  transformer: (options: TOptions) => DataTransformer;
+  transformer: (options: TOptions, data: DataFrame[]) => Observable<DataFrame[]>;
```

Comment:
is there any value in the independent `DataTransformer` interface?  if not, this should be renamed so that it is the action `transform` rather than the thing that will transform

Reply:
Sorry, not sure I follow

Reply:
Before `transformer(options)` returned a function (DataTransformer) that could process DataFrame[]'s

Now the registry item holds the function directly and it takes data as an input -- i think that is fine, but should be named a verb rather than a noun.

"transformer" is the name of a thing... a thing that transforms!
"transform" is an action on data

previously "transformer" returned a function, now it actually processes the data, so i think

Reply:
Gotcha and I agree that makes more sense.

Reply:
Renamed it to operator as that is what is is now :)